content/articles/understanding-embeddings-in-machine-learning/index.md

```diff
@@ -0,0 +1,98 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /understanding-embeddings-in-machine-learning/
+title: Understanding embeddings in Machine Learning
+description: This article will look at embeddings in detail and solve real-world problems involving complex data.
+author: lilian-ogoti
+date: 2022-01-29T00:00:00-07:00
+topics: [Machine Learning]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/understanding-embeddings-in-machine-learning/hero.jpeg
+    alt: Understanding embeddings in Machine Learning hero image
+---
+Handling non-numerical data poses challenges to data scientists since most machine learning models tend to prefer working with numbers. 
+<!--more-->
+Data that contains words or sentences, graphs, lists, and probability distributions tend to be more complex to deal with when it comes to machine learning. However, it is impossible to avoid such data in the real world and hence needs first to be transformed into [vector space](https://machinelearningmastery.com/a-gentle-introduction-to-vector-space-models/).
+
+Existing approaches are not helpful and applicable as they tend to have flaws when transforming the data. A better approach would be to use [embedding](https://towardsdatascience.com/neural-network-embeddings-explained), a low dimensional representation of high dimensional data. The article will look at embeddings in detail and solve real-world problems involving complex data.
+
+### The concept of embedding in data transformation
+The concept of embeddings is highly dependent on machine learning models. To grasp the concept better, we must first know the requirements of the machine learning models. One of the most common features of machine learning algorithms is that they only accept low-dimensional data as their inputs. Each input feature in a neural network is a number meaning that non-numeric variables must be converted into numbers and vectors.  
+
+For instance, variables such as users and items in [recommender systems](https://www.analyticssteps.com/blogs/what-are-recommendation-systems-machine-learning) are non-numeric. However, we can use the product ID to represent the items. However, to achieve this, we have to treat the ID as the continuous numeric variables the way it is done in neural networks. It would imply that the higher numbers are more significant than the lower numbers. The same happens with similar numbers, assuming the same numbers represent the same items. The analogy makes sense for variables like age. However impossible to represent [categorical variables](https://www.sciencedirect.com/topics/mathematics/categorical-variable). Before using embeddings, one of the commonly used methods to represent categorical variables was called [one-hot encoding](https://machinelearningmastery.com/how-to-one-hot-encode-sequence-data-in-python/), as discussed in the following section.
```

Comment:
```suggestion
For instance, variables such as users and items in [recommender systems](https://www.analyticssteps.com/blogs/what-are-recommendation-systems-machine-learning) are non-numeric. We can use the product ID to represent the items. However, to achieve this, we have to treat the ID as the continuous numeric variables the way it is done in neural networks. It would imply that the higher numbers are more significant than the lower numbers. 
The same happens with similar numbers, assuming the same numbers represent the same items. The analogy makes sense for variables like age. However impossible to represent [categorical variables](https://www.sciencedirect.com/topics/mathematics/categorical-variable). Before using embeddings, one of the commonly used methods to represent categorical variables was [one-hot encoding](https://machinelearningmastery.com/how-to-one-hot-encode-sequence-data-in-python/), as discussed in the following section.
```